signerverifier/rsa.go

```diff
@@ -113,21 +118,20 @@ func LoadRSAPSSKeyFromFile(path string) (*SSLibKey, error) {
 	}
 
 	switch k := keyObj.(type) {
-	case *rsa.PublicKey:
-		pubKeyBytes, err := x509.MarshalPKIXPublicKey(k)
-		if err != nil {
-			return nil, fmt.Errorf("unable to load RSA key from file: %w", err)
-		}
-		key.KeyVal.Public = strings.TrimSpace(string(generatePEMBlock(pubKeyBytes, PublicKeyPEM)))
-
-	case *rsa.PrivateKey:
-		pubKeyBytes, err := x509.MarshalPKIXPublicKey(k.Public())
-		if err != nil {
-			return nil, fmt.Errorf("unable to load RSA key from file: %w", err)
-		}
-		key.KeyVal.Public = strings.TrimSpace(string(generatePEMBlock(pubKeyBytes, PublicKeyPEM)))
-		key.KeyVal.Private = strings.TrimSpace(string(generatePEMBlock(pemData.Bytes, RSAPrivateKeyPEM)))
+		case *rsa.PublicKey, *rsa.PrivateKey:
+			pubKeyBytes, err := marshalAndGeneratePEM(k)
+			if err != nil {
+				return nil, fmt.Errorf("unable to load RSA key from file: %w", err)
+			}
+			key.KeyVal.Public = strings.TrimSpace(string(pubKeyBytes))
+
+			if _, ok := k.(*rsa.PrivateKey); ok {
+				key.KeyVal.Private = strings.TrimSpace(string(generatePEMBlock(pemData.Bytes, RSAPrivateKeyPEM)))
+			}
+		default :
+			return nil, fmt.Errorf("unexpected key type: %T", k)
```

Comment:
Small thought, if you move this to `marshalAndGeneratePEM`, you could remove this switch statement and have just one.

i.e.
```
call marshalAndGeneratePEM
// error handle ^
Public <- TrimSpace
if ... {
  Private <- TrimSpace
}
```

Reply:
I have pulled the switch case into the helper function